### PR TITLE
fix(cli): Create user fails with --force-update

### LINKED ISSUE #19139: createuser: Error when trying to --force-update user
`createuser` fails when run against an existing user. 

```
$ sudo docker-compose run --rm web sentry --version
sentry, version 10.1.0.dev0 (0bf9ffa08ff2)
$ sudo docker-compose run --rm web sentry createuser --email testuser@test.com --password pass1 --no-superuser --no-input --force-update                                                                  
21:32:17 [WARNING] sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured.
21:32:21 [INFO] sentry.plugins.github: apps-not-configured
User created: testuser@test.com
Added to organization: sentry
$ sudo docker-compose run --rm web sentry createuser --email testuser@test.com --password pass2 --no-superuser --no-input --force-update                                                                  
21:33:46 [WARNING] sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured.
21:33:49 [INFO] sentry.plugins.github: apps-not-configured
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/__init__.py", line 166, in main
    cli(prog_name=get_prog(), obj={}, max_content_width=100)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/decorators.py", line 30, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/commands/createuser.py", line 83, in createuser
    user.save(force_update=force_update)
  File "/usr/local/lib/python2.7/site-packages/sentry/models/user.py", line 141, in save
    return super(User, self).save(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/contrib/auth/base_user.py", line 80, in save
    super(AbstractBaseUser, self).save(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 896, in _save_table
    raise ValueError("Cannot force an update in save() with no primary key.")
ValueError: Cannot force an update in save() with no primary key.
```

### PR DESCRIPTION
**Pull request by BYK**
Fixes #19139.

### CODE REVIEW

File: src/sentry/runner/commands/createuser.py [hunk 1]
```diff
@@ -76,8 +76,14 @@ def createuser(email, password, superuser, no_password, no_input, force_update):
...
-    if User.objects.filter(username=email).exists():
+    try:
+        existing_user_id = User.objects.get(username=email).id
+    except User.DoesNotExist:
+        existing_user_id = None
+
+    if existing_user_id is not None:
         if force_update:
+            user.id = existing_user_id
             user.save(force_update=force_update)
```
Comment (wedamija) [inline src/sentry/runner/commands/createuser.py:87]: I don't think this is what you want to do:

```
In [25]: user_3 = User.objects.create(username='test_tesjtst', email='test@test.com')
In [28]: user_3.set_password('hello')
In [30]: (user_3.username, user_3.email, user_3.password)
Out[30]:
('test_tesjtst',
 'test@test.com',
 'pbkdf2_sha256$36000$pK6zZ1CGK7Qu$Pcm9GpEINVg5lJGL/SDh0k9qR9qnpqcIQdTXMivad6U=')

In [31]: some_user = User(email='hi@bye.com', username='hi@by.com')
In [32]: some_user.id = user_3.id
In [33]: some_user.save()

In [34]: user_3.refresh_from_db()
In [35]: (user_3.username, user_3.email, user_3.password)
Out[35]: ('hi@by.com', 'hi@bye.com', '')
```

This overwrites all parameters on the existing row with what's in the new `User`, which means nulling out anything that you didn't specify. You should probably instead user `existing_user.update(<fields you want to update>)`

Comment (BYK) [reply src/sentry/runner/commands/createuser.py]: Okay, refactored around this.

**Review by wedamija — APPROVED** [review]

File: src/sentry/runner/commands/createuser.py [hunk 2]
```diff
@@ -69,23 +69,27 @@ def createuser(email, password, superuser, no_password, no_input, force_update):
...
         if force_update:
-            user.save(force_update=force_update)
-            click.echo(f"User updated: {email}")
+            user.update(**fields)
+            verb = "updated"
         else:
             click.echo(f"User: {email} exists, use --force-update to force")
             sys.exit(3)
     else:
+        user = User(**fields)
```
Comment (wedamija) [inline src/sentry/runner/commands/createuser.py:90]: You could just use `User.objects.create` here as well

Comment (BYK) [reply src/sentry/runner/commands/createuser.py:90]: Ah, didn't know about this one. I thought Python was the language where there was only one obvious way to do a certain thing? 🤣

**Pull request merged**